--- firefox_profile.py  2021-09-28 19:46:21.935430200 +0800
+++ firefox_profile.py.bak      2021-09-29 17:48:03.389054500 +0800
@@ -79,12 +79,12 @@
             shutil.copytree(self.profile_dir, newprof,
                             ignore=shutil.ignore_patterns("parent.lock", "lock", ".parentlock"))
             self.profile_dir = newprof
-            os.chmod(self.profile_dir, 0o755)
+            os.chmod(self.profile_dir, 0o700)
             self._read_existing_userjs(os.path.join(self.profile_dir, "user.js"))
         self.extensionsDir = os.path.join(self.profile_dir, "extensions")
         self.userPrefs = os.path.join(self.profile_dir, "user.js")
         if os.path.isfile(self.userPrefs):
-            os.chmod(self.userPrefs, 0o644)
+            os.chmod(self.userPrefs, 0o600)

     # Public Methods
     def set_preference(self, key, value):
