--- webdriver.py        2021-09-28 19:46:22.029176900 +0800
+++ webdriver.py.bak    2021-09-29 17:47:22.369080100 +0800
@@ -52,6 +52,7 @@
                  executable_path="geckodriver", options=None,
                  service_log_path="geckodriver.log", firefox_options=None,
                  service_args=None, desired_capabilities=None, log_path=None,
+                 port=0,
                  keep_alive=True):
         """Starts a new local session of Firefox.

@@ -158,7 +159,7 @@
         if capabilities.get("marionette"):
             capabilities.pop("marionette")
             self.service = Service(
-                executable_path,
+                executable_path, port=port,
                 service_args=service_args,
                 log_path=service_log_path)
             self.service.start()

