--- webdriver.py	2022-03-10 09:00:03.035218032 +0000
+++ webdriver.py.bak	2022-03-10 09:05:34.673872208 +0000
@@ -52,6 +52,7 @@
                  executable_path="geckodriver", options=None,
                  service_log_path="geckodriver.log", firefox_options=None,
                  service_args=None, desired_capabilities=None, log_path=None,
+                 port=0,
                  keep_alive=True):
         """Starts a new local session of Firefox.
 
@@ -158,7 +159,7 @@
         if capabilities.get("marionette"):
             capabilities.pop("marionette")
             self.service = Service(
-                executable_path,
+                executable_path, port=port,
                 service_args=service_args,
                 log_path=service_log_path)
             self.service.start()
